grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/query/ReasonerQueryImpl.java

```diff
@@ -123,6 +129,17 @@
                 .build();
     }
 
+    private Set<IsaAtom> inferEntityTypes() {
```

Comment:
Maybe this can return the `Stream` to begin with? I ask because this returns a `Set` but in `getVarTypeMap` it gets converted to a stream again.